filesystems/native.py

```diff
@@ -59,14 +59,19 @@ def _remove_file(fs, path):
         raise
 
 
-def _create_directory(fs, path, with_parents):
+def _create_directory(fs, path, with_parents, allow_existing):
     try:
         if with_parents:
-            os.makedirs(str(path))
+            if _PY3:
+                os.makedirs(str(path), exist_ok=allow_existing)
+            else:
+                os.makedirs(str(path))
         else:
             os.mkdir(str(path))
     except (IOError, OSError) as error:
         if error.errno == exceptions.FileExists.errno:
+            if allow_existing and fs.is_dir(path):
```

Comment:
Hm -- does the stdlib also have this TOCTOU?

(the call to `is_dir` specifically is race condition happy)

Reply:
I thought the same thing when I wrote it...  but failed to stick a `TODO:` there to make sure it got caught.  Thanks.  I'll look it over.

Reply:
Well what do you know...

https://github.com/python/cpython/blob/v3.8.1/Lib/os.py#L222-L226
```python3
    except OSError:
        # Cannot rely on checking for EEXIST, since the operating system
        # could give priority to other errors like EACCES or EROFS
        if not exist_ok or not path.isdir(name):
            raise
```